### PR TITLE
Renames TSHttpTxnSetHttpRetStatus to TSHttpTxnStatusSet

### PR DESCRIPTION
**Pull request by zwoop**
This also adds a new API, TSHttpTxnStatusGet(TSHttpTxn txnp),
which is the corresponding Getter for this functionality.

### CODE REVIEW

**Comment by randall** [comment]
[approve ci autest]

File: proxy/api/ts/ts.h [hunk 1]
```diff
@@ -1524,8 +1524,8 @@ tsapi TSReturnCode TSVConnArgIndexReserve(const char *name, const char *descript
 tsapi TSReturnCode TSVConnArgIndexNameLookup(const char *name, int *arg_idx, const char **description);
 tsapi TSReturnCode TSVConnArgIndexLookup(int arg_idx, const char **name, const char **description);
 
-/* ToDo: This is a leftover from olden days, can we eliminate? */
-tsapi void TSHttpTxnSetHttpRetStatus(TSHttpTxn txnp, TSHttpStatus http_retstatus);
+tsapi void TSHttpTxnStatusSet(TSHttpTxn txnp, TSHttpStatus status);
+tsapi TSHttpStatus TSHttpTxnStatusGett(TSHttpTxn txnp);
```
Comment (d2r) [inline proxy/api/ts/ts.h]: Spelled correctly?

Comment (zwoop) [reply proxy/api/ts/ts.h]: No, nice catch!

**Comment by SolidWallOfCode** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
Cherry-picked to 8.0.x

**Pull request merged**